irods/test/user_group_test.py

```diff
@@ -418,6 +440,55 @@ def test_group_admin_can_create_users__issue_428(self):
             if rodsuser:
                 admin.users.remove(rodsuser.name)
 
+    # Test to ensure a rodsuser cannot lock themselves out by corrupting the catalog entry containing their
+    # encoded password.
+
+    # Need to record a separate issue for this.
+
+    def test_against_database_password_corruption_in_modify_password__issue_XYZ(self):
+        ses = helpers.make_session()
+        if ses.users.get( ses.username ).type != 'rodsadmin':
+            self.skipTest( 'Only a rodsadmin may run this test.')
+
+        user_alice = None
+        extra="."
+        NEWPASS='1234567890123456789012345'
+        OLDPASS='123456789012345678901234'
+
+        try:
+            user_alice = ses.users.create('alice', 'rodsuser')
+            ses.users.modify('alice', 'password', OLDPASS)
+
+            def alice_login(pw):
+                with iRODSSession(user='alice', password=pw, host=ses.host, port=ses.port, zone=ses.zone) as alice:
+                    home = helpers.home_collection( alice )
+                    alice.collections.get(home)
+
+            alice_login(OLDPASS)
+
+            # Alice makes an unsuccessful attempt at changing her own password.
+            with iRODSSession(user='alice', password=OLDPASS, host=ses.host, port=ses.port, zone=ses.zone) as alice:
+                me = alice.users.get(alice.username)
+                try:
+                    me.modify_password(OLDPASS+extra, NEWPASS)
+                except (ex.CAT_PASSWORD_ENCODING_ERROR, Bad_password_change_parameter):
+                    pass
+
+            # Either the new or old password should succeed.
+            successes = 0
+            for password in (OLDPASS, NEWPASS):
+                try:
+                    alice_login(password)
+                except ex.CAT_INVALID_AUTHENTICATION:
+                    pass
+                else:
+                    successes += 1
+            self.assertGreater(successes, 0)
```

Comment:
And this just proves that alice can still authenticate and do things? The only thing is that alice doesn't know which password is valid.

Is that correct?

Reply:
The idea is actually that, in some instances prior to this fix, neither password was valid because the db entry got scrambled, and not in a good ("recoverable") way.

Reply:
Post-this-fix, the bad scrambling will not happen because we are no longer leaving parameter filtering up to the server; we're doing it in the client

Reply:
Got it. I really dislike we can't be sure which password is in affect.

Reply:
An comment explaining that the reason we do it this way is because we don't know which password is active could be helpful.